### ISSUE TITLE
Different sigmoid_focal_loss value between mmdet1.0 and mmdet2.2.

### ISSUE THREAD

**Issue by thisisi3** [open-mmlab/mmdetection#3228]
The 'sigmoid_focal_loss' in mmdet.models.losses.focal_loss produces different values between mmdet1.0(483beb71) and mmdet2.2(25ac8f50). Here is my test case:

pred = torch.tensor([-1.0, 1.0], device='cuda:0')
tar  = torch.tensor([1,1], device='cuda:0')
weight = torch.tensor([1.0, 1.0], device='cuda:0')
reduction='mean'
avg_factor=2
print(focal_loss.sigmoid_focal_loss(pred.view(-1, 1), tar, weight,
                                        gamma=2.0, alpha=0.25,
                                        reduction=reduction, avg_factor=avg_factor))

mmdet1.0: tensor(0.0906, device='cuda:0')
mmdet2.2: tensor(0.2717, device='cuda:0')

Strangely when I set alpha==0.5, they produce the same value, in this case.

**Comment by thisisi3** [comment]
Later I realize that sigmoid_focal_losst should produce different values because now background label is num_class instead of 0. 
This also means we can not use FocalLoss in RPN. I have been trying FocalLoss in RPN recently and did not work, I think this is the reason.

**Comment by hust-nj** [comment]
> Later I realize that sigmoid_focal_losst should produce different values because now background label is num_class instead of 0.
> This also means we can not use FocalLoss in RPN. I have been trying FocalLoss in RPN recently and did not work, I think this is the reason.

Does it mean the focal loss in mmdet1.0 use background label as _0_ but in mmdet 2.0 use background label as _num_class_?
@thisisi3
